geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/EnumeratedHistogram.scala

```diff
@@ -0,0 +1,51 @@
+/***********************************************************************
+  * Copyright (c) 2013-2015 Commonwealth Computer Research, Inc.
+  * All rights reserved. This program and the accompanying materials
+  * are made available under the terms of the Apache License, Version 2.0 which
+  * accompanies this distribution and is available at
+  * http://www.opensource.org/licenses/apache2.0.php.
+  *************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.collection.mutable
+
+case class EnumeratedHistogram[T](attribute: String) extends Stat {
+
+  val map: scala.collection.mutable.HashMap[T, Long] = mutable.HashMap[T, Long]()
+
+  override def observe(sf: SimpleFeature): Unit = {
+    val sfval = sf.getAttribute(attribute)
+
+    if (sfval != null) {
+      sfval match {
+        case tval: T =>
+          updateMap(tval, 1)
+      }
+    }
+  }
+
+  override def toJson(): String = {
+    s"[${map.map{ case (key: T, count: Long) => s"$key:$count" }.mkString(",")}]"
+  }
+
+  override def add(other: Stat): Stat = {
+    other match {
+      case eh: EnumeratedHistogram[T] =>
+        combine(eh)
+        this
```

Comment:
@elahrvivaz @anthonyccri I tried to fix this here: https://github.com/locationtech/geomesa/compare/master...jnh5y:wip_stats_hlist

Anthony's point gets at the point that the 'add' method should take the type of the implementing class.  With implementations like 
```
case class MinMax[T <: Comparable[T]](attribute: String) extends Stat[MinMax[T]]
```
we can achieve that.  

The sticky wicket comes in with the implementation of SeqStat.  In order to implement, 'add' there, we'd need to be able to make a List[Monoid] a Monoid again (by delegating to coordinate-wise addition).

I struggled with that for a bit, and I'm happy to hear suggestions.  In my experimentation, I poked at Scalaz and Shapeless, but I haven't seen any clever options.  

One possibility is a SeqStat may have to retain a list of types (somehow), so that each stat can be cast appropriately during add.